index.js

```diff
@@ -89,22 +90,22 @@ exports.execute = async (srcName, query, params = {}, options = {}) => {
 
     console.debug(
       `Oracle Adapter: Connection secured: ${process.hrtime(start)[0]}s ${
-        process.hrtime(start)[1] / 1000000
+      process.hrtime(start)[1] / 1000000
       }ms`
     );
-    const result = await conn.execute(query, params, options);
+    result = await conn.execute(query, params, options);
 
     console.debug(
       `Oracle Adapter: Query executed: ${process.hrtime(start)[0]}s ${
-        process.hrtime(start)[1] / 1000000
+      process.hrtime(start)[1] / 1000000
       }ms`
     );
-
-    await conn.close();
-    return result;
   } catch (err) {
     console.error("Oracle Adapter: Error while executing query", err);
     throw new Error(err.message);
+  } finally {
+    await conn.close();
+    return result;
```

Comment:
This will not pass the no-unsafe-finally lint rule.

We can still return from the try block, but have `await conn.close()` in the finally block and this will act as expected.  The return from the try block will be held until the finally block completes.